migrations/consolidatefeatures_migration.go

```diff
@@ -0,0 +1,29 @@
+package migrations
+
+import (
+	"bytes"
+
+	cctypes "github.com/loomnetwork/go-loom/builtin/types/chainconfig"
+)
+
+func ConsolidateFeaturesMigration(ctx *MigrationContext) error {
+	// Pull data from ChainConfig
+	_, chainconfigCtx, err := resolveChainConfig(ctx)
+	if err != nil {
+		return err
+	}
+	featuresFromState := ctx.state.Range([]byte(featurePrefix))
```

Comment:
In case feature key does not exists, only then you directly set feature key with feature struct.

Reply:
Done.